packages/utils/types/index.d.ts

```diff
@@ -1,42 +1,7 @@
 // Definitions by: Junyoung Clare Jang <https://github.com/Ailrun>
 // TypeScript Version: 2.2
-
-export interface RegisteredCache {
-  [key: string]: string
-}
-
-export interface StyleSheet {
-  container: HTMLElement
-  nonce?: string
-  key: string
-  insert(rule: string): void
-  flush(): void
-  tags: Array<HTMLStyleElement>
-}
-
-export interface EmotionCache {
-  inserted: {
-    [key: string]: string | true
-  }
-  registered: RegisteredCache
-  sheet: StyleSheet
-  key: string
-  compat?: true
-  nonce?: string
-  insert(
-    selector: string,
-    serialized: SerializedStyles,
-    sheet: StyleSheet,
-    shouldCache: boolean
-  ): string | void
-}
-
-export interface SerializedStyles {
-  name: string
-  styles: string
-  map?: string
-  next?: SerializedStyles
-}
+import type { SerializedStyles } from '@emotion/serialize'
+import type { RegisteredCache, EmotionCache } from '@emotion/cache'
```

Comment:
We can't import from other packages here because we don't depend on them. This is also true at type level - even if we still don't import anything at runtime.

I'd keep the existing types here (to avoid breaking changes) but I'd deprecate them in favor of the proper ones exported by specific packages). In the future, we should just remove `export` from those local types here. This way we'll be able to depend on TS structural type system without introducing any dependencies to this package.